### PR TITLE
RAS-2242: Handle Large DATA Segments

### PR DESCRIPTION
**Pull request by straylightrun**
# Description
Our modified FCSParser library can write FCS files, but we are not following theFCS file standard which states that "When any portion of a segment falls outside the 99,999,999 byte limit, '0's are substituted in the HEADER for that segments begin and end byte offset." 
[FCS Standard](https://www.bioconductor.org/packages/release/bioc/vignettes/flowCore/inst/doc/fcs3.html)

This resulted in an additional byte being added to the header when writing the byte offset for the `DATA` segment. That additional byte corrupted the reading of the following `TEXT` segment which meant that FlowJo, R, and Python could not read the FCS file.
 
## Issues Fixed
[RAS-2242]
 
## Reviewer Instructions
Please merge for testing in Develop. 
_NOTE_: Most of the changes are due to Black formatting

## Type of Change
- [x] Bug Fix
- [ ] New Feature(s)
- [ ] Refactor
- [ ] Tech Debt
- [ ] Testing
- [ ] Breaking Change
- [ ] Documentation
 
## How Has This Been Tested
- Loaded the unlabeled `Well_E02.fcs` from screen `8822` using this library
- Added metadata manually using json stored in S3 for this well
- Saved as a new FCS file using this library and verified that it could be read by FlowJo and that new metadata had been added
 
# Checklist
- [ ] New tests added for new features and/or breaking changes
- [x] Upstream changes in destination branch have been merged/rebased
- [x] Comments added to new code, especially in hard to understand blocks
- [ ] Documentation updated if applicable
- [x] New code has been appropriately styled/linted

[RAS-2242]: https://notablelabs.atlassian.net/browse/RAS-2242

### CODE REVIEW

File: fcsparser/api.py [hunk 1]
```diff
@@ -418,78 +460,125 @@ def get_channel_names(self):
...
-        return b''.join([self.version(),
-                         self.header_space_characters,
-                         str(self.text_offset_start).rjust(self.header_offset_block_length).encode('utf-8'),
-                         str(self.text_offset_end).rjust(self.header_offset_block_length).encode('utf-8'),
-                         str(self.data_offset_start).rjust(self.header_offset_block_length).encode('utf-8'),
-                         str(self.data_offset_end).rjust(self.header_offset_block_length).encode('utf-8'),
-                         str(self.analysis_offset_start).rjust(self.header_offset_block_length).encode('utf-8'),
-                         str(self.analysis_offset_end).rjust(self.header_offset_block_length).encode('utf-8'),
-                         BYTE_SEP])
+        # check for data start/end outside of maximum
```
Comment (straylightrun) [inline fcsparser/api.py:493]: This is where the relevant changes begin

Comment (lawrlee) [reply fcsparser/api.py:493]: Are there any other changes we should merge from upstream?

Comment (straylightrun) [reply fcsparser/api.py:493]: There are a few in the original repo, but I didn't want to risk breaking anything.

**Review by lawrlee — APPROVED** [review]
LGTM. Can you tag and push a new version to gemfury?

**Pull request merged**